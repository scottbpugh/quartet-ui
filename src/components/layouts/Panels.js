// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import React, {Component} from "react";
import {connect} from "react-redux";
import {FormattedMessage} from "react-intl";
import {loadPageTitle} from "../../reducers/layout";
import "react-resizable/css/styles.css";
import {ResizableBox} from "react-resizable";
import "./panels.css";

/**
 * LeftPanel
 *
 * @param {object} props Takes a title and children elems.
 *
 * @return {ReactElement} The left panel element.
 */
class _LeftPanel extends Component {
  render() {
    return (
      <ResizableBox className="left-panel" axis="x" width={300}>
        {/* <div className="left-panel"> */}
        <h4 className="left-panel-title pt-dark">
          {/* We use a new message from passed props because Redux uses plain objects. */}
          <FormattedMessage {...this.props.pageTitle} />
        </h4>
        <div
          style={{
            overflowX: "hidden",
            overflowY: "visible",
            whiteSpace: "nowrap"
          }}>
          {this.props.children}
        </div>
        {/* </div> */}
      </ResizableBox>
    );
  }
}

export const LeftPanel = connect(
  (state, ownProps) => {
    return {
      pageTitle: state.layout.pageTitle
    };
  },
  {loadPageTitle}
)(_LeftPanel);

/**
 * RightPanel
 *
 * @param {object} props Children elements.
 *
 * @return {ReactElement} The right panel element.
 */
class _RightPanel extends Component {

  loadTitle() {
    if (typeof this.props.title === "string") {
      this.props.loadPageTitle({id: "plugins.unimplemented.noTitle", defaultMessage: this.props.title});
    } else {
      this.props.loadPageTitle({...this.props.title.props});
    }
  }
  
  componentDidMount() {
    // this.props.loadPageTitle(this.props.title.props.id);
    this.loadTitle(this.props.title);
  }
 
  render() {
    return (
      <div className="right-panel">
        <div ref="rightPanel">{this.props.children}</div>
      </div>
    );
  }
}

export const RightPanel = connect(
  (state, ownProps) => {
    return {
      pageTitle: state.layout.pageTitle,
      currentPath: state.layout.currentPath
    };
  },
  {loadPageTitle}
)(_RightPanel);

/**
 * Default - function returning a left/right panel layout.
 *
 *
 * @param {Object} props title/leftPanel/rightPanel jsx
 *
 * @return {ReactElement} The layout element
 */
class _Panels extends Component {
  render() {
    const clonedChildren = this.props.children.map((child, index) => {
      if (index === 1) {
        // Add a unique currentPath key to force unmounting of right panel
        // whenever the path changes.
        const newProps = {...child.props, key: this.props.currentPath};
        return React.cloneElement(child, newProps);
      }
      return child;
    });
    return <div className="main-container">{clonedChildren}</div>;
  }
}

export const Panels = connect(
  (state, ownProps) => {
    return {
      pageTitle: state.layout.pageTitle,
      currentPath: state.layout.currentPath
    };
  },
  {loadPageTitle}
)(_Panels);
