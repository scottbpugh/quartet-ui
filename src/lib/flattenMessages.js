// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

export function flattenMessages(nestedMessages, prefix = "") {
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key];
    const prefixedKey = prefix ? `${prefix}.${key}` : key;

    if (typeof value === "string") {
      messages[prefixedKey] = value;
    } else {
      Object.assign(messages, flattenMessages(value, prefixedKey));
    }

    return messages;
  }, {});
}

window.qu4rtet.exports("lib/flattenMessages", this);
