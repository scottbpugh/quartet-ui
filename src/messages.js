// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

export default {
  "en-US": {
    app: {
      nav: {
        servers: "Servers",
        numberRange: "Number Ranges",
        dashboard: "Dashboard",
        server: "Server"
      },
      serverSettings: {
        serverSettings: "Server Settings",
        addAServer: "Add a New Server"
      }
    },
    plugins: {
      numberRange: {
        pool: "Pool",
        region: "Region",
        allocation: "Allocation",
        allocateButton: "Allocate from Pool",
        createdOn: "Created On",
        readableName: "Readable Name",
        machineName: "Machine Name",
        status: "Status",
        requestThreshold: "Request Threshold",
        regions: "Regions",
        numberRangePools: "Number Range Pools",
        active: "active",
        inactive: "inactive",
        range: "Range",
        to: "to",
        state: "State",
        addRegion: "Add a New Region",
        addPool: "Add a New Pool",
        noRegionInPool: "There is currently no region in this pool.",
        regionDetailTitle: "Pool {poolName} Regions"
      }
    }
  },
  "fr-FR": {
    app: {
      nav: {
        servers: "Serveurs",
        numberRange: "Plage de numéros",
        dashboard: "Tableau de bord",
        server: "Serveur"
      },
      serverSettings: {
        serverSettings: "Paramètres serveur",
        addAServer: "Ajouter un serveur"
      }
    },
    plugins: {
      numberRange: {
        pool: "Groupe",
        region: "Région",
        allocation: "Attribution",
        allocateButton: "Attribuer au groupe",
        createdOn: "Créé le",
        readableName: "Nom",
        machineName: "Nom machine",
        status: "Statut",
        requestThreshold: "Seuil de requêtes",
        regions: "Régions",
        numberRangePools: "Groupes de plages de numéros",
        active: "actif",
        inactive: "inactif",
        range: "Plage",
        to: "à",
        state: "État",
        addRegion: "Ajouter une région",
        addPool: "Ajouter un Groupe",
        noRegionInPool: "Il n'existe aucune region dans ce groupe.",
        regionDetailTitle: "Regions du groupe {poolName}"
      }
    }
  }
};
