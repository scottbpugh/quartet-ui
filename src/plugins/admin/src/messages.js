export default {
  "en-US": {plugins: {admin: {}}},
  "fr-FR": {
    plugins: {
      admin: {}
    }
  }
};
