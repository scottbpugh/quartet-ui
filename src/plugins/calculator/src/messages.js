// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

const defaultLocaleMsg = {
  gtinCalculator: {
    GTINCalculatorTitle: "GTIN Calculator",
    calcPlaceholder: "Enter an NDC code",
    indicatorDigit: "Indicator {indicator}",
    GTINs: "GTINs"
  }
};

const french = {
  GTINCalculatorTitle: "Calculatrice de GTIN",
  calcPlaceholder: "Entrer un code NDC",
  indicatorDigit: "Indicator {indicator}",
  GTINs: "GTINs"
};

export default {
  "en-US": {plugins: {...defaultLocaleMsg}},
  "fr-FR": {
    plugins: {
      gtinCalculator: {
        ...defaultLocaleMsg.gtinCalculator,
        ...french
      }
    }
  }
};
