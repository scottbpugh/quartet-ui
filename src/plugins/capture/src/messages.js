// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

const defaultLocaleMsg = {
  capture: {
    rulesTopNav: "Rules",
    tasksTopNav: "Tasks",
    addRule: "Add a New Rule",
    editRule: "Edit Existing Rule",
    addStep: "Add a New Step",
    editStepParam: "Edit Step Parameter",
    editStep: "Edit Step",
    addTask: "Create a New Task",
    deleteStep: "Delete Step",
    deleteStepConfirm: "Are you sure you want to delete this step?",
    deleteRule: "Delete Rule",
    deleteRuleConfirm: "Are you sure you want to delete this rule?",
    uploadFile: "File Upload",
    uploadFileTitle: "{ruleName} File Upload",
    createTask: "Create Task with File",
    uploadedFile: "File uploaded for rule {ruleName}",
    addRuleParam: "Add a New Rule Parameter",
    addStepParam: "Add a New Step Parameter",
    processFileError:
      "An error occurred while processing this file. Please check the QU4RTET server for more information. {error}",
    errorFetchRuleParams:
      "An error occurred while retrieving parameters for this rule. Please check the QU4RTET server for more information.",
    ruleParameters: "Rule Parameters",
    stepParameters: "Step Parameters",
    addRuleParameter: "Add Rule Parameter",
    editRuleParam: "Edit Rule Parameter",
    noRuleFound: "No Capture module detected on server",
    steps: "Steps",
    taskCreated: "Task with id {taskID} has been created.",
    tasksTotal: "{tasksCount} tasks total.",
    selectRule: "Select a Rule",
    taskDetail: "Task Detail",
    restart: "Restart",
    confirmRestart: "Are you sure you want to restart this task?",
    confirmRestartBody:
      "Restarting this task will re-execute all of the logic for this rule.",
    executeTaskError:
      "An error occurred while attempting to re-excute this task: {error}"
  }
};

const french = {
  rulesTopNav: "Règles",
  tasksTopNav: "Tâches",
  addRule: "Ajouter une nouvelle règle",
  editRule: "Éditer règle",
  addStep: "Ajouter une nouvelle étape",
  editStep: "Éditer étape",
  editStepParam: "Éditer un paramètre d'étape",
  addTask: "Créer une nouvelle tâche",
  deleteStep: "Supprimer une étape",
  deleteStepConfirm: "Etes-vous certain de vouloir supprimer cette étape ?",
  deleteRule: "Supprimer une règle",
  deleteRuleConfirm: "Etes-vous certain de vouloir supprimer cette règle ?",
  uploadFile: "Envoi de fichier",
  uploadFileTitle: "Envoi de fichier {ruleName}",
  createTask: "Créer une tâche avec fichier",
  uploadedFile: "Fichier envoyé pour règle {ruleName}",
  addRuleParam: "Ajouter un nouveau paramètre de règle",
  addStepParam: "Ajouter un nouveau paramètre d'étape",
  processFileError:
    "Une erreur est survenue lors du traitement de ce fichier. Veuillez vérifier le serveur QU4RTET pour plus d'informations.",
  errorFetchRuleParams:
    "Une erreur s'est produite lors de la récupération des paramètres pour cette règle. Veuillez vérifier le serveur QU4RTET pour plus d'informations.",
  ruleParameters: "Paramètres de règle",
  addRuleParameter: "Ajouter un paramètre de règle",
  editRuleParam: "Editer un paramètre de règle",
  noRuleFound: "Module de capture non détecté sur ce serveur.",
  taskCreated: "Une tâche avec l'ID {taskID} a été créée.",
  tasksTotal: "Total des tâches: {tasksCount}",
  steps: "Étapes",
  selectRule: "Sélectionner une règle",
  restart: "Recommencer"
};

export default {
  "en-US": {plugins: {...defaultLocaleMsg}},
  "fr-FR": {
    plugins: {
      capture: {
        ...defaultLocaleMsg.Capture,
        ...french
      }
    }
  }
};
