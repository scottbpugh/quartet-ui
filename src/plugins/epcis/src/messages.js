// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

const defaultLocaleMsg = {
  epcis: {
    topNavItem: "Event Data",
    addEvent: "Add Event",
    editEvent: "Edit Event",
    addEntry: "Add Entry",
    editEntry: "Edit Entry",
    aggregationEvent: "Aggregation Event",
    objectEvent: "Object Event",
    transactionEvent: "Transaction Event",
    transformationEvent: "Transformation Event",
    what: "What",
    eventType: "Event Type",
    eventID: "Event ID",
    parentID: "Parent ID",
    epcList: "EPC List",
    inputEPCList: "Input EPC List",
    outputEPCList: "Output EPC List",
    ILMD: "Instance/Lot Master Data",
    childQuantityList: "Child Quantity List",
    inputQuantityList: "Input Quantity List",
    outputQuantityList: "Output Quantity List",
    quantityList: "Quantity List",
    childEPCs: "Child EPCs",
    businessTransactionList: "Business Transaction List",
    eventsTotal: "{eventsCount} events total",
    entriesTotal: "{entriesCount} entries total",
    noEPCISFound: "No EPCIS module detected on server.",
    aggregationEvents: "Aggregation Events",
    objectEvents: "Object Events",
    transactionEvents: "Transaction Events",
    transformationEvents: "Transformation Events",
    events: "Events",
    entries: "Entries",
    errorLoadingEvent: "An error occurred while loading this event. {error}",
    errorLoadingEntry: "An error occurred while loading this entry. {error}",
    errorLoadingEntries:
      "An error occurred while loading entries from this server. {error}",
    errorLoadingEvents:
      "An error occurred while loading events from this server. {error}",
    errorLoadingEntryGeo:
      "An error occurred while loading the geographic data related to this entry. {error}",
    where: "Where",
    why: "Why",
    when: "When",
    objectEvent: "Object Event",
    entryList: "Entries"
  }
};

const french = {
  topNavItem: "Données d'événements",
  addEvent: "Ajouter un évènement",
  editEvent: "Modifier l'événement",
  addEntry: "Ajouter une entrée",
  editEntry: "Modifier l'entrée",
  aggregationEvent: "Événement d'agrégation",
  objectEvent: "Événement d'objet",
  transactionEvent: "Événement de transaction",
  transformationEvent: "Événement de transformation",
  what: "Quoi",
  when: "Quand",
  where: "Où",
  why: "Pourquoi",
  eventType: "Type d'événement",
  eventID: "ID d'événement",
  parentID: "ID du parent",
  epcList: "Liste EPC",
  inputEPCList: "Liste EPC d'entrée",
  outputEPCList: "Liste EPC de sortie",
  ILMD: "Données de base d'instance/lot",
  childQuantityList: "Child Quantity List",
  inputQuantityList: "Input Quantity List",
  outputQuantityList: "Output Quantity List",
  quantityList: "Quantity List",
  childEPCs: "Child EPCs",
  businessTransactionList: "Business Transaction List",
  eventsTotal: "Total: {eventsCount}",
  entriesTotal: "Total: {entriesCount}",
  noEPCISFound: "Module EPCIS non détecté sur ce serveur.",
  aggregationEvents: "Événements d'agrégation",
  objectEvents: "Événements d'objet",
  transactionEvents: "Événements de transaction",
  transformationEvents: "Événements de transformation",
  events: "Événements",
  entries: "Entrées",
  entryList: "Entrées",
  errorLoadingEvent:
    "Une erreur s'est produite lors du chargement de cet événement. {error}",
  errorLoadingEntry:
    "Une erreur s'est produite lors du chargement de cette entrée. {error}",
  errorLoadingEntries:
    "Une erreur s'est produite lors du chargement des entrées de ce serveur. {error}",
  errorLoadingEvents:
    "Une erreur s'est produite lors du chargement des événements de ce serveur. {error}",
  errorLoadingEntryGeo:
    "Une erreur s'est produite lors du chargement des données géographiques liées à cettre entrée. {error}"
};

export default {
  "en-US": {plugins: {...defaultLocaleMsg}},
  "fr-FR": {
    plugins: {
      epcis: {
        ...defaultLocaleMsg.epcis,
        ...french
      }
    }
  }
};
