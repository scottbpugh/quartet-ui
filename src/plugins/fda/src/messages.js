// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

const defaultLocaleMsg = {
  fda: {
    FDALookup: "FDA Lookup",
    manufacturerName: "Manufacturer Name",
    brandName: "Brand Name",
    genericName: "Generic Name",
    packageNdc: "Package NDC",
    productNdc: "Product NDC",
    effectiveDate: "Effective Date",
    generateTradeItem: "Generate Trade Item",
    generateTradeItems: "Generate Trade Item(s)",
    fdaProvidedNDC: "The FDA-provided NDC",
    indicatorDigit: "Indicator Digit",
    next: "next",
    previous: "previous"
  }
};

const french = {
  FDALookup: "Recherche FDA",
  manufacturerName: "Nom du fabricant",
  brandName: "Marque",
  genericName: "Nom générique",
  packageNdc: "NDC paquet",
  productNdc: "NDC produit",
  effectiveDate: "Date effective",
  generateTradeItem: "Créer un article commercial",
  generateTradeItems: "Créer article commerciaux",
  fdaProvidedNDC: "NDC FDA",
  indicatorDigit: "Chiffre indicateur",
  next: "suivant",
  previous: "précedent"
};

export default {
  "en-US": {plugins: {...defaultLocaleMsg}},
  "fr-FR": {
    plugins: {
      fda: {
        ...defaultLocaleMsg.fda,
        ...french
      }
    }
  }
};
