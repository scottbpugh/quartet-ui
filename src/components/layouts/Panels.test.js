// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React from "react";
import renderer from "react-test-renderer";
import {Panels, LeftPanel, RightPanel} from "./Panels";

import {IntlProvider} from "react-intl";
import {addLocaleData, FormattedMessage} from "react-intl";
import messages from "messages";
import en from "react-intl/locale-data/en";
import fr from "react-intl/locale-data/fr";
import {flattenMessages} from "lib/flattenMessages";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

addLocaleData([...en, ...fr]);
let locale = "en-US";
const middlewares = [thunk];

const initialState = {
  dashboard: {notifications: []},
  serversettings: {},
  intl: {
    defaultLocale: "en-US",
    locale: locale,
    messages: flattenMessages(messages[locale])
  },
  layout: {pageTitle: {id: "app.nav.servers"}},
  plugins: {}
};

const mockStore = configureStore(middlewares);

it("renders nested components correctly", () => {
  let store = mockStore(initialState);
  const panels = renderer
    .create(
      <IntlProvider
        locale={locale}
        messages={initialState.intl.messages}
        store={store}>
        <Panels store={store}>
          <LeftPanel store={store}>
            <h1>Something</h1>
          </LeftPanel>
          <RightPanel
            title={<FormattedMessage id="app.nav.servers" />}
            store={store}>
            <h2>Something Else</h2>
          </RightPanel>
        </Panels>
      </IntlProvider>
    )
    .toJSON();
  expect(panels).toMatchSnapshot();
});
