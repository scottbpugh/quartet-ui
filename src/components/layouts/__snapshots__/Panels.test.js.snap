// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nested components correctly 1`] = `
<div>
  <div
    className="main-container"
  >
    <div
      className="left-panel react-resizable"
      style={
        Object {
          "height": "undefinedpx",
          "width": "300px",
        }
      }
    >
      <h4
        className="left-panel-title bp3-dark"
      >
        <span>
          Servers
        </span>
      </h4>
      <div
        style={
          Object {
            "overflowX": "hidden",
            "overflowY": "visible",
            "whiteSpace": "nowrap",
          }
        }
      >
        <h1>
          Something
        </h1>
      </div>
      <span
        className="react-resizable-handle"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "touchAction": "none",
          }
        }
      />
    </div>
    <div
      className="right-panel"
    >
      <div>
        <h2>
          Something Else
        </h2>
      </div>
    </div>
  </div>
</div>
`;
